test/scale/scale_test.go

```diff
@@ -0,0 +1,197 @@
+			Command:  "make",
+			Args:     []string{"e2e-bootstrap"},
+			StartLog: "Installing Azure Disk CSI Driver...",
+			EndLog:   "Azure Disk CSI Driver installed",
+		}
+
+		createMetricsSVC := testtypes.TestCmd{
+			Command:  "make",
+			Args:     []string{"create-metrics-svc"},
+			StartLog: "create metrics service ...",
+			EndLog:   "metrics service created",
+		}
+		if !*skipClusterBootstrap {
+			testutil.ExecTestCmd([]testtypes.TestCmd{e2eBootstrap, createMetricsSVC})
+		}
+	}
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	// Default storage driver configuration is CSI. Freshly built
+	// CSI driver is installed for that case.
+	if testconsts.IsTestingMigration || testconsts.IsUsingInTreeVolumePlugin {
+		cmLog := testtypes.TestCmd{
+			Command:  "bash",
+			Args:     []string{"test/utils/controller-manager-log.sh"},
+			StartLog: "===================controller-manager log=======",
+			EndLog:   "===================================================",
+		}
+		testutil.ExecTestCmd([]testtypes.TestCmd{cmLog})
+	}
+
+	if testconsts.IsTestingMigration || !testconsts.IsUsingInTreeVolumePlugin {
+		checkPodsRestart := testtypes.TestCmd{
+			Command:  "bash",
+			Args:     []string{"test/utils/check_driver_pods_restart.sh", "log"},
+			StartLog: "Check driver pods if restarts ...",
+			EndLog:   "Check successfully",
+		}
+		testutil.ExecTestCmd([]testtypes.TestCmd{checkPodsRestart})
+
+		os := "linux"
+		cloud := "azurepubliccloud"
+		if testconsts.IsWindowsCluster {
+			os = "windows"
+		}
+		if testconsts.IsAzureStackCloud {
+			cloud = "azurestackcloud"
+		}
+		createExampleDeployment := testtypes.TestCmd{
+			Command:  "bash",
+			Args:     []string{"hack/verify-examples.sh", os, cloud},
+			StartLog: "create example deployments",
+			EndLog:   "example deployments created",
+		}
+		testutil.ExecTestCmd([]testtypes.TestCmd{createExampleDeployment})
+
+		azurediskLogArgs := []string{"test/utils/azuredisk_log.sh", "azuredisk"}
+		if testconsts.IsUsingCSIDriverV2 {
+			azurediskLogArgs = append(azurediskLogArgs, "v2")
+		}
+
+		azurediskLog := testtypes.TestCmd{
+			Command:  "bash",
+			Args:     azurediskLogArgs,
+			StartLog: "===================azuredisk log===================",
+			EndLog:   "===================================================",
+		}
+
+		deleteMetricsSVC := testtypes.TestCmd{
+			Command:  "make",
+			Args:     []string{"delete-metrics-svc"},
+			StartLog: "delete metrics service...",
+			EndLog:   "metrics service deleted",
+		}
+
+		e2eTeardown := testtypes.TestCmd{
+			Command:  "make",
+			Args:     []string{"e2e-teardown"},
+			StartLog: "Uninstalling Azure Disk CSI Driver...",
+			EndLog:   "Azure Disk CSI Driver uninstalled",
+		}
+
+		if *skipClusterBootstrap {
+			testutil.ExecTestCmd([]testtypes.TestCmd{azurediskLog})
+		} else {
+			testutil.ExecTestCmd([]testtypes.TestCmd{azurediskLog, deleteMetricsSVC, e2eTeardown})
+		}
+
+		if !testconsts.IsTestingMigration && !testconsts.IsUsingCSIDriverV2 {
```

Comment:
The scale tests do not need to install or uninstall the driver from the `install-driver.sh` and `uninstall-driver.sh` scripts, nor create the example deployments again.